Videoplayer Crash if no preferred tuner set

fuzzard opened the pull request:
I came across this issue, when you havent set a preferred tuner in settings, the videoplayer will crash in the ss.good() loop. Ive just added check if preferred tuner is empty (none set) the function will continue on. Im not sure on the best way to continue, so feel free to provide any alternative solution.
With this change, the Videoplayer will start and run the channel as intended

fuzzard committed:
Videoplayer Crash if no preferred tuner set

I came across this issue, when you havent set a preferred tuner in settings, the videoplayer will crash in the ss.good() loop. Ive just added check if preferred tuner is empty (none set) the function will continue on. Im not sure on the best way to continue, so feel free to provide any alternative solution.
With this change, the Videoplayer will start and run the channel as intended

MatthewLundberg commented:
Good catch.  I can't believe I didn't hit this condition.

Your fix is good, but I see a better path.  First, change the loop conditional from `while(ss.good())` to `while(ss.good() && !ss.eof())`.  Second, add a test to immediately return `nullptr` in the case that the string is empty (note that `nullptr` will be returned if the set `ids` is empty).  I'll implement these changes soon.

fuzzard committed:
Update Info.cpp

A change like the following? Happy to leave it to you if you want, but didnt really want to provide problems without attempting a solution.

MatthewLundberg commented:
I just pushed a fix for this to my master branch.

fuzzard commented:
Thanks, ill close this PR
